x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -261,7 +264,13 @@ export class EnterpriseSearchPlugin implements Plugin {
     if (!config.ui?.enabled) {
       return;
     }
-    const { cloud, share } = plugins;
+    const { cloud, share, licensing } = plugins;
+
+    let isEnterpriseLicense: boolean = false;
```

Comment:
if we're going to track this with a variable I think we should add it to the class like you did for `licenseSubscription` instead of having it be a scoped variable to the `setup` function.

Reply:
We already have licensing variables available in the Enterprise Search plugin so we shouldn't really need a new one anyway.

Reply:
I changed this part so that license checking is done inside the subscribe